pkg/storage/bucket.go

```diff
@@ -5,33 +5,92 @@ import (
 	"context"
 	"crypto/md5"
 	"fmt"
+	"github.com/kelseyhightower/envconfig"
 	"github.com/minio/minio-go/v6"
 	"github.com/sirupsen/logrus"
+	"github.com/virtual-vgo/vvgo/pkg/log"
 	"github.com/virtual-vgo/vvgo/pkg/tracing"
 	"mime"
 	"net/http"
 	"net/url"
+	"path/filepath"
 	"strings"
 	"time"
 )
 
-type Object struct {
-	ContentType string
-	Tags        Tags
-	Buffer      bytes.Buffer
+const ProtectedLinkExpiry = 24 * 3600 * time.Second // 1 Day for protect links
+
+var logger = log.Logger()
+
+type Warehouse struct {
+	config      Config
+	minioClient *minio.Client
 }
 
-type Tags map[string]string
+type Config struct {
+	Minio MinioConfig `envconfig:"minio"`
+}
 
-func NewObject(mediaType string, buffer *bytes.Buffer) *Object {
-	return &Object{
-		ContentType: mediaType,
-		Buffer:      *buffer,
+func (x *Config) ParseEnv() error {
+	return envconfig.Process("storage", x)
+}
+
+type MinioConfig struct {
+	Enabled   bool
+	Endpoint  string `default:"localhost:9000"`
+	Region    string `default:"sfo2"`
+	AccessKey string `default:"minioadmin"`
+	SecretKey string `default:"minioadmin"`
+	UseSSL    bool   `default:"false"`
+}
+
+func NewWarehouse(config Config) (*Warehouse, error) {
+	client := Warehouse{config: config}
+	if config.Minio.Enabled {
+		var err error
+		client.minioClient, err = minio.New(config.Minio.Endpoint, config.Minio.AccessKey, config.Minio.SecretKey, config.Minio.UseSSL)
```

Comment:
Right now, the default minio config has minio disabled, which means by default, bucket operations are nooped. I feel like this will lead confusing to some confused debugging. Instead, this builder should require _either_ minio or noop.

I could add an inmem cache here as well, but like... there's no use for it right now.

Reply:
Nooped it. Default noop is false. If not nooped and a bad minio config is supplied, we get an error.